webapp/src/routes/game/[game_id]/+page.server.ts

```diff
@@ -0,0 +1,241 @@
+import type { PageServerLoad } from './$types';
+import { supabase } from "$lib/supabaseClient.js";
+
+export const load: PageServerLoad = async ({ params, fetch }) => {
+    const gameId = params.game_id; // Get game_id from the URL
+    const gameData = await fetchGame(gameId);
+
+    if (!gameData) {
+        console.error('Game not found');
+        return { filteredQueryData: [] };
+    }
+
+    const categoryData = await fetchCategory(gameData.category_id);
+
+    if (!categoryData) {
+        console.error('Category not found');
+        return { filteredQueryData: [] };
+    }
+
+    // TODO: Add logic here to fetch the right amount of cards based on what is already there.
+    // Also the cards need to be "handed" to the players equally in the beginning.
+    const numberOfCards = 5;
+
+    const cards = await generateCards(gameId, categoryData.name, gameData.category_id, numberOfCards, gameData.difficulty, fetch);
+
+    return { filteredQueryData: cards };
+};
+
+// Function to fetch game data based on game_id
+async function fetchGame(gameId: string): Promise<Game | null | undefined> {
+    const { data, error } = await supabase
+        .from('Game') // Use the correct table name
+        .select(`
+            id,
+            status,
+            max_card_count,
+            difficulty,
+            category_id,
+            whose_turn_id
+        `)
+        .eq('id', gameId)
+        .single(); // Ensure you get a single record
+
+    if (error || !data) {
+        console.error('Error fetching game:', error);
+        return null;
+    }
+
+    return data as Game;
+}
+
+// Function to fetch category based on category_id
+async function fetchCategory(categoryId: string): Promise<Category | null> {
+    const { data, error } = await supabase
+        .from('Category') // Use the correct table name
+        .select(`
+            id,
+            name,
+            picture_path,
+            api_route,
+            hex_color
+        `)
+        .eq('id', categoryId)
+        .single(); // Ensure you get a single record
+
+    if (error || !data) {
+        console.error('Error fetching category:', error);
+        return null;
+    }
+
+    return data as Category;
+}
+
+
+// Function to generate cards based on the category
+async function generateCards(gameId: string, categoryName: string | null, categoryId: string, limit: number, difficulty: string, fetch: typeof globalThis.fetch): Promise<Card[]> {
+    let sparqlQuery = '';
+
+    // Select the SPARQL query based on the category
+    switch (categoryName) {
```

Comment:
Create API routes and use the category's api_route field to distinguish what logic is required to retrieve the data.